### PR TITLE
Fix 'publish docker' stage in Jenkinsfile

### PR DESCRIPTION
**Pull request by egvimo**
Updating the CI-CD generator Jenkinsfile.

- This fixes the node install bug for Gradle
- use prod builds
- and use the same commands described in [the docs](https://www.jhipster.tech/docker-compose/#3).

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/ci-cd/templates/jenkins/Jenkinsfile.ejs [hunk 1]
```diff
@@ -184,10 +184,10 @@ node {
 <%= gitLabIndent %>        // A pre-requisite to this step is to setup authentication to the docker registry
     <%_ if (buildToolMaven) { _%>
 <%= gitLabIndent %>        // https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#authentication-methods
-<%= gitLabIndent %>        sh "./mvnw -ntp jib:build"
+<%= gitLabIndent %>        sh "./mvnw -ntp package -Pprod verify jib:build"
```
Comment (pascalgrimaud) [inline generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:187]: I wonder if we can skip tests at this phase ?

Comment (egvimo) [reply generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:187]: You're right, I'll remove that part.

File: generators/ci-cd/templates/jenkins/Jenkinsfile.ejs [hunk 2]
```diff
@@ -184,10 +184,10 @@ node {
 <%= gitLabIndent %>        // A pre-requisite to this step is to setup authentication to the docker registry
     <%_ if (buildToolMaven) { _%>
 <%= gitLabIndent %>        // https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#authentication-methods
-<%= gitLabIndent %>        sh "./mvnw -ntp jib:build"
+<%= gitLabIndent %>        sh "./mvnw -ntp package -Pprod jib:build"
     <%_ } else if (buildToolGradle) { _%>
 <%= gitLabIndent %>        // https://github.com/GoogleContainerTools/jib/tree/master/jib-gradle-plugin#authentication-methods
-<%= gitLabIndent %>        sh "./gradlew jib"
+<%= gitLabIndent %>        sh "./gradlew bootJar jib -Pprod -PnodeInstall --no-daemon"
```
Comment (mshima) [inline generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:190]:
```suggestion
<%= gitLabIndent %>        sh "./gradlew bootJar jib -Pprod --no-daemon"
```

nodeInstall should be removed for consistency with everything else.
Please add it commented to gradle.properties:
```

## Install and use a local version of node and npm for consistency when using different machines.
# nodeInstall
```

You can propose to use by default but it should be another PR.

Comment (egvimo) [reply generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:190]: But it is used in (almost) all Gradle tasks here in Jenkins pipeline. See e.g. backed or frontend tests.

Comment (mshima) [reply generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:190]: Your are right.
This can lead to different behaviors in different situations, but since this used in other places it's ok.

Can you add the config to gradle.properties anyway?

Comment (mshima) [reply generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:190]: Thanks.

Comment (egvimo) [reply generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:190]: I've added the section to the properties.

And yes, I already have encountered some problems with this, because the default npm version of the LTS node is 6.14 and the specified version in the Gradle build file is 7.8. ☹️

Comment (mshima) [reply generators/ci-cd/templates/jenkins/Jenkinsfile.ejs:190]: That should be tracked by https://github.com/jhipster/generator-jhipster/issues/14249.

**Comment by mshima** [comment]
We have https://github.com/jhipster/generator-jhipster/issues/13531 open to decide which command to use for docker.
It should be used here too.

**Comment by egvimo** [comment]
> We have #13531 open to decide which command to use for docker.
> It should be used here too.

Done.

**Comment by egvimo** [comment]
> We have https://github.com/jhipster/generator-jhipster/issues/13531 open to decide which command to use for docker.
> It should be used here too.

@mshima The discussion is now closed, but I don't quite know what the outcome is. Should I use `compile` and update the PR?

**Comment by mshima** [comment]
@egvimo I think the correct was to use package or compile, but since the PR is not merged and the documentation still uses verify, you should use verify for now.

You will probably need to updated test snapshots for tests to pass.

**Comment by egvimo** [comment]
@mshima I have updated the command and the tests.

**Review by mshima — APPROVED** [review]
LGTM

**Comment by mshima** [comment]
@pascalgrimaud are you already doing do release?

**Comment by pascalgrimaud** [comment]
Everything is ready on my computer but I'll do it this afternoon.
So this one can be merged, as it looks safe
